rbac/providers/common/inbound_filters.py

```diff
@@ -71,3 +75,34 @@ def inbound_value_filter(inbound_value, field_name):
             epoch_zero = datetime.datetime(1970, 1, 1, tzinfo=value.tzinfo)
             value = int((value - epoch_zero).total_seconds())
     return value
+
+
+def outbound_queue_filter(entry_data, provider):
+    """ Filters inbound_queue 'data' field to match outbound_queue's
+    'data' field. If an attribute is missing from entry_data, the
+    attribute will be set to either an empty list or empty string.
+    This function currently only supports comparing AD group changes.
+
+    Args:
+        entry_data:
+            obj:    A dict containing a RethinkDB role entry.
+        provider: (str) A string indicating identity provider to filter with.
+            Expecting value of either 'azure' or 'ldap'.
+    Returns:
+        result: (dict) Returns the filtered inbound_queue data with the
+            proper fields to compare against outbound_queue's data field.
+    Throws:
+        TypeError: Throws when provider is not of value 'azure' or 'ldap'
+    """
+    if provider not in ("azure", "ldap"):
```

Comment:
We can remove provider parameter as it is no longer needed for this function (make sure to update docstrings accordingly too).